packages/shared/components/modals/AliasSelector.svelte

```diff
@@ -0,0 +1,39 @@
+<script lang="typescript">
+    import { ActivityType, IAliasActivityData, selectedAccountActivities } from '@core/wallet'
+    import { Modal, Text, TextType } from 'shared/components'
+    import { truncateString } from 'shared/lib/helpers'
+    import { get } from 'svelte/store'
+    import { fade } from 'svelte/transition'
+
+    export let modal: Modal = undefined
+    export let selected: string = undefined
+    export let onClose: () => void
+
+    $: aliasIds = get(selectedAccountActivities)
+        .filter((activity) => activity.type === ActivityType.Alias)
+        .map((activity) => {
+            const aliasId = (activity.data as IAliasActivityData).aliasId
+            return { value: aliasId, label: truncateString(aliasId, 9, 9) }
```

Comment:
Why do you get the `aliasId` from the `selectedAccountActivities`? It feels like this could error, what if you transferred control over an alias output already. It would be part of your activities, but you would still need to create one. I think it's better to get this info from wallet.rs.

Reply:
good point!